src/sources/kubernetes_logs/pod_metadata_annotator.rs

```diff
@@ -56,10 +58,15 @@ impl PodMetadataAnnotator {
     /// [`FILE_KEY`] field set with a file that the line came from.
     pub fn annotate(&self, event: &mut Event, file: &str) -> Option<()> {
         let log = event.as_mut_log();
-        let uid = parse_log_file_path(file)?.pod_uid;
-        let guard = self.pods_state_reader.get(uid)?;
+        let file_info = parse_log_file_path(file)?;
+        let guard = self.pods_state_reader.get(file_info.pod_uid)?;
         let entry = guard.get_one()?;
         let pod: &Pod = entry.as_ref();
+
+        log.insert(
+            &self.fields_spec.container_name,
+            file_info.container_name.to_owned(),
+        );
         annotate_from_metadata(log, &self.fields_spec, &pod.metadata);
```

Comment:
We might want to rename this `annotate_from_pod_metadata`, since we're not also annotating from non-pod metadata, which happens outside this function.

Reply:
That's not really a concern here - there are no other candidates that have metadata in this code. What's specializes this fn here is the fact that it reads from the metadata. We'll add other fns in the future - for instance, to populate the container image we'd need one to `annotate_from_spec`. In all cases `Pod` is assumed here.

The mod name itself might need to be altered, but then it can be considered as mentioning "metadata" in a broader sense.

Either way, I don't think this PR should be doing this.